crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -47,46 +47,40 @@ pub struct AzdlsProfile {
 }
 
 const DEFAULT_ENDPOINT_SUFFIX: &str = "dfs.core.windows.net";
+lazy_static::lazy_static! {
+    static ref DEFAULT_AUTHORITY_HOST: Url = Url::parse("https://login.microsoftonline.com").expect("Default authority host is a valid URL");
+}
 
 impl AzdlsProfile {
     /// Validate the Azure storage profile.
     ///
     /// # Errors
-    /// - Fails if the bucket name is invalid.
-    /// - Fails if the region is too long.
-    /// - Fails if the key prefix is too long.
-    /// - Fails if the region or endpoint is missing.
-    /// - Fails if the endpoint is not a valid URL.
-    pub async fn validate(
-        &mut self,
-        credential: Option<&AzCredential>,
-    ) -> Result<(), ValidationError> {
-        // If key_prefix is provided, remove any trailing and leading slashes.
+    /// - Fails if the filesystem name is invalid.
+    /// - Fails if the key prefix is too long or invalid.
+    /// - Fails if the account name is invalid.
+    /// - Fails if the endpoint suffix is invalid.
+    pub(super) fn normalize(&mut self) -> Result<(), ValidationError> {
+        validate_filesystem_name(&self.filesystem)?;
+        self.endpoint_suffix = normalize_endpoint_suffix(self.endpoint_suffix.take())?;
+        self.normalize_key_prefix()?;
+        validate_account_name(&self.account_name)?;
+
+        Ok(())
+    }
+
+    fn normalize_key_prefix(&mut self) -> Result<(), ValidationError> {
         if let Some(key_prefix) = self.key_prefix.as_mut() {
             *key_prefix = key_prefix.trim_matches('/').to_string();
         }
 
-        let AzdlsProfile {
-            filesystem,
-            key_prefix,
-            account_name: _,
-            authority_host: _,
-            endpoint_suffix,
-            sas_token_validity_seconds: _,
-        } = self;
-
-        if endpoint_suffix.as_deref().is_some_and(|s| s.contains('/')) {
-            return Err(ValidationError::InvalidProfile {
-                source: None,
-                reason: "Storage Profile `endpoint_suffix` must not contain slashes.".to_string(),
-                entity: "EndpointSuffix".to_string(),
-            });
+        if let Some(key_prefix) = self.key_prefix.as_ref() {
+            if key_prefix.is_empty() {
+                self.key_prefix = None;
+            }
```

Comment:
let's maybe move these fns a bit down in the impl block, I tend to read files top to bottom and I find it easier to understand what's going on if the fns are ordered from high abstraction to lower level fns

Reply:
done